### DISCUSSION TITLE
KernelCI rootfs

### DISCUSSION THREAD

**Discussion by aistcv** [kernelci/kernelci-core#1426] [Q&A]
From experiments it is observed that when using [rootfs.ext4.xz](https://github.com/kernelci/kernelci-core/blob/main/config/rootfs/debos/rootfs.yaml#L127) on FVP, we are having a stripped image (with apt and dpkg missing). 

Taking a closer look over  kernelci-core and the history of commits it seems that:
	• The intention was to have a full debian system in rootfs.ext4.xz  when the functionality was introduced ("[646a8199  Add three new images with the full debian system](https://github.com/kernelci/kernelci-core/commit/646a8199b46b7fa287cfdabe2e1fc1c701ad7207)")
	• That all changed with "[f68edce0  config/rootfs/debos: fix ext4 corrupt images](https://github.com/kernelci/kernelci-core/commit/f68edce0dcb3c23cf64e85a460229d4070861770)". It seems that the move and compression actions on the image happen before the image is unmounted, which happens after _rootfs.yaml_ is completed. The solution was to run those with _postprocess: true_ which will run those after the filesystem is unmounted, but
	• At that point [scripts/crush.sh](https://github.com/kernelci/kernelci-core/blob/main/config/rootfs/debos/scripts/crush.sh) would have run and stripped the rootfs.

We need rootfs with apt and dpkg because, the newly built [bullseye-kselftest](https://github.com/kernelci/kernelci-core/blob/main/config/core/rootfs-configs.yaml#L159) rootfs when deployed in lava job the below blocker is observed.
```
./kselftest.sh: 266: apt-get: not found
./kselftest.sh: 269: apt-get: not found
ERROR: Failed to install dependencies, exiting...
```

Any recommendation to prevent removing the packages before unmounting is much appreciated. 
Or should we build uncrushed images as well?

**Comment by nuclearcat** [comment]
I wonder if we just comment **package_management**, build by hand and compare rootfs size difference.
If it is insignificant we can apply such changes to whole KernelCI, then i think it is completely reasonable to keep apt-get in rootfs.

Another approach might be to check kselftest.sh, why it is calling apt-get, probably it detects that some software/tool missing. If it is small package, required by kselftests - we might just add it in build recipe.

**Reply by broonie** [reply]
The use of apt in the kselftest test definition just looks pointless, it's xz-utils and something similarly trivial IIRC. Frankly if it was something large that'd be even more reason why we shouldn't be doing that given that the direct usage of apt in the test definition ends up bypassing cache configuration for the lab unless you use a transparent proxy.
